src/main/java/org/broadinstitute/hellbender/transformers/SimpleRepeatMaskTransformer.java

```diff
@@ -0,0 +1,55 @@
+package org.broadinstitute.hellbender.transformers;
+
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+/**
+ * Masks read bases with a supra-threshold number of A/T's or G/C's within a given window size.
+ */
+public class SimpleRepeatMaskTransformer implements ReadTransformer {
+    public static final long serialVersionUID = 1L;
+    private static final byte MASKED_BASE_QUAL = 2;
+
+    final int windowSize, threshAT, threshGC;
+
+    public SimpleRepeatMaskTransformer(final int threshAT, final int threshGC, final int windowSize) {
+        this.threshAT = threshAT;
+        this.threshGC = threshGC;
+        this.windowSize = windowSize;
+    }
+
+    @Override
+    public GATKRead apply(final GATKRead read) {
```

Comment:
A suggestion for how to implement it. 

I'll start with what my understanding is:
This function, rolls along the provided sequence with a fixed-size window in a base--by-base fashion, calculates the AT/GC (with N) content within the window and mask out the bases and changes the qual accordingly if GC content is high or low.

Now my suggestion:
Since there are many reads to be checked against, I would imagine this could be a hot spot. So making it fast is probably a worthy thing to do:
* instead of summing up base counts for each window, why not store a `previousATNcount` and `previousGCNcount`, considering that each time we step into a new window, these two variables are going to change only +/-1. This would get rid of the `basesCount` array;
* as you have already did in the current implementation, the changes in bases count would be +/-1, why not, instead of setting bases and quals in each window, simply mark if the current window defined by `start` and `end` should be masked, and finally maximally extend the `start` and `end`, then go change those bases and quals
* in `maskWindow`, instead of closed interval, why not make it semi-open [start, end)? this would also get rid of some of the "-1"'s in the `apply()` function.